libpod/runtime_volume_linux.go

```diff
@@ -164,6 +167,85 @@ func (r *Runtime) newVolume(options ...VolumeCreateOption) (_ *Volume, deferredE
 	return volume, nil
 }
 
+// UpdateVolumePlugins reads all volumes from all configured volume plugins and
+// imports them into the libpod db. It also checks if existing libpod volumes
+// are removed in the plugin, in this case we try to remove it from libpod.
+// On errors we continue and try to do as much as possible. all errors are
+// returned as array in the returned struct.
+// This function has many race conditions, it is best effort but cannot guarantee
+// a perfect state since plugins can be modified from the outside at any time.
+func (r *Runtime) UpdateVolumePlugins(ctx context.Context) *define.VolumeReload {
```

Comment:
Feels like this should return (*define.VolumeReload, error) with a single overall error for things like database errors that would dead-stop the process (e.g. line 212)

Reply:
Well at line 212 we potentially already added new volumes. If you want to return an error there it would hide the result. I also do not want to call `r.state.AllVolumes()` at the beginning since this would allow more time for potential race conditions.

Reply:
Fair enough